api_app/analyzers_manager/classes.py

```diff
@@ -47,15 +48,31 @@ def init_report_object(self) -> AnalyzerReport:
         """
         Returns report object set in *start* fn
         """
-        return AnalyzerReport.objects.create(
-            job_id=self.job_id,
-            analyzer_name=self.analyzer_name,
-            report={},
-            errors=[],
-            status=AnalyzerReport.Statuses.PENDING.name,
-            runtime_configuration=self.kwargs.get("runtime_conf", {}),
-            task_id=self.kwargs["task_id"],
+        # unique constraint ensures only one report is possible
+        _report_qs = AnalyzerReport.objects.filter(
+            job_id=self.job_id, analyzer_name=self.analyzer_name
         )
+        if _report_qs.exists():  # case: recurring analyzer run
+            _report_qs.update(
+                report={},
+                errors=[],
+                status=AnalyzerReport.Statuses.PENDING.name,
+                # runtime_configuration stays the same
+                task_id=self.kwargs["task_id"],
+                start_time=timezone.now(),
+                end_time=timezone.now(),
+            )
+            return _report_qs[0]
+        else:
+            return AnalyzerReport.objects.create(
+                job_id=self.job_id,
+                analyzer_name=self.analyzer_name,
+                report={},
+                errors=[],
+                status=AnalyzerReport.Statuses.PENDING.name,
+                runtime_configuration=self.kwargs.get("runtime_conf", {}),
+                task_id=self.kwargs["task_id"],
+            )
```

Comment:
Can't we use the `get_or_create` or `get_or_update` method instead ? what do u think ?

Reply:
The logic duplication of `init_report_object` method in the 2 subclasses would be redundant after #583.

Reply:
anyways I have to update an existing object, `get_or_create` would only be useful if I had to avoid duplicating some object

Reply:
well, I haven't checked out on `update_or_create` yet, looks like it should do it.